builtin/game/item.lua

```diff
@@ -38,6 +38,10 @@ function core.get_pointed_thing_position(pointed_thing, above)
 end
 
 function core.dir_to_facedir(dir, is6d)
+	-- nil check for dir (for use with place_node)
+	if not dir then
+		return 0
```

Comment:
@tenplus1 I think the reason for this is that placing programmatically creates an empty "placer" object that has no look direction, so ``placer:get_look_dir()`` returns nil.
I think it's better to either either make the placer have a direction after: https://github.com/minetest/minetest/blob/1c1c97cbd1d7913ac12bf550ec02c97f843a0fd3/src/script/cpp_api/s_item.cpp#L72

Or check that the direction can be null in: https://github.com/minetest/minetest/blob/b24e6433df3c3b2926568aff9c0173459e3e8eab/builtin/common/misc_helpers.lua#L386